src/compiler/utilities.ts

```diff
@@ -4642,26 +4642,48 @@ namespace ts {
         return identifier.length >= 3 && identifier.charCodeAt(0) === CharacterCodes._ && identifier.charCodeAt(1) === CharacterCodes._ && identifier.charCodeAt(2) === CharacterCodes._ ? identifier.substr(1) : identifier;
     }
 
-    export function levenshtein(s1: string, s2: string): number {
-        let previous: number[] = new Array(s2.length + 1);
-        let current: number[] = new Array(s2.length + 1);
-        for (let i = 0; i < s2.length + 1; i++) {
+    const CHANGE_COST = 2; // Make substitutions costly, since when they appear in short strings they're normally obvious errors we don't want to promote
+    const INSERT_COST = 1;
+    const DELETE_COST = 1;
+    const TRANSPOSE_COST = 0.95; // Transpositions slightly cheaper than insertion/deletion/substitution to promote transposed errors
+    export const MIN_LEV_DIST = TRANSPOSE_COST;
+    export function dameraulevenshtein(s1: string, s2: string): number {
+        if (s1.length > s2.length) [s1, s2] = [s2, s1]; // Swap arguments so s2 is always the short one to use minimal memory

```

Comment:
We were already reusing our arrays to save memory (and complicating the code a little to do so), so while not go all the way and use the minimal memory for the algorithm?